expyfun/visual/_visual.py

```diff
@@ -826,6 +826,87 @@ def __init__(self, ec, colors=('w', 'k')):
         self.set_radius(1, 1, units='pix')
 
 
+class ProgressBar(object):
+    """A progress bar that can be displayed between sections
+
+    This uses two rectangles, one outline, and one solid to show how much
+    progress has been made in the experiment.
+
+    Parameters
+    ----------
+    ec : instance of ExperimentController
+        Parent EC.
+    pos : array-like
+        4-element array-like with X, Y center and width, height.
+    units : str
+        Units to use. These will apply to all spatial aspects of the drawing.
+        Must be either ''norm'' or ''pix''.
+    colors : list or tuple of matplotlib Colors
+        Colors to fill and outline the bar respectively. Defaults to green and
+        white.
+    text : bool
+        Whether to show the percent done as text below the bar. If ''True'',
+        leave room on the screen below the bar
+
+    Returns
+    -------
+    bar : instance of ProgressBar
+        The progress bar.
+    """
+    def __init__(self, ec, pos, units='norm', colors=('g', 'w'), text=False):
+        self._ec = ec
+        if len(colors) != 2:
+            raise ValueError('colors must have length 2')
+        if units != 'norm' and units != 'pix':
+            raise ValueError('units must be either ''norm'' or ''pix''')
+
+        pos = np.array(pos)
+        self._pos = pos
+        self._width = pos[2]
+        self._units = units
+
+        # initialize the bar with zero progress
+        self._pos_bar = pos.copy()
+        self._pos_bar[0] -= self._width * 0.5
+        self._init_x = self._pos_bar[0]
+        self._pos_bar[2] = 0
+
+        self._rectangles = [Rectangle(ec, self._pos_bar, units, colors[0],
+                                      None),
+                            Rectangle(ec, self._pos, units, None, colors[1])]
+        if units == 'norm':
+            self._text_pos = [pos[0], pos[1] - pos[3] * 0.5 - 0.1]
```

Comment:
I feel like there has to be a more eloquent way to do this. Especially for using units in pixels.